E.X.P.E.R.I-MENTOR fixes

GunHog opened the pull request:
Fixes an issue in which manually ejecting an item from the
E.X.P.E.R.I-MENTOR caused it to become inoperable.
EDIT: More bugfixes added:
- Relic discovery works again!
- Fixed a runtime when the Experimentor tries to throw items at players.
- Ejected items are no longer sent to the void.
- If the machine is deconstructed, it will also eject the loaded item, if any.

Commit:
Fixes ejection issue with the E.X.P.E.R.I-MENTOR

Fixes an issue in which manually ejecting an item from the
E.X.P.E.R.I-MENTOR caused it to become inoperable.

HoaxIT commented:
:+1: Looks good

GunHog commented:
I found two other bugs with the machine that I am going to tack on here. (Ejected items sometimes getting lost in the 0,0,0 void and relic discover option never working)

Incoming5643 commented:
if this is turning into a general Experimentor fix pull, last I checked if you disassemble the machine while it's loaded, the loaded item is lost.

Commit:
INCREASES SANITY!!

- Fixes a bug in which ejected items are sent to a null turf.
- Added a sanity check when the machine tries to throw a null object.
- Relic discovery works again!

Commit:
Deconstruction fix

When deconstructed, the Experimentor will now spit out any item it
contains, should it have one.

GunHog commented:
@Incoming5643 bug noted and fixed.

AnonymousNow commented:
If I'm not too late to the party, last I checked the EXPERIMENTOR disassociates itself with the research network if both it and a deconstructive analyser is linked in a small vicinity.
